httpx/_transports/asgi.py

```diff
@@ -24,6 +31,10 @@ def create_event() -> "Event":
         return asyncio.Event()
 
 
+async def async_byte_iterator(bytestring: bytes) -> AsyncIterator[bytes]:
+    yield bytestring
```

Comment:
I'd agree that having this built-in supported by HTTPCore would be beneficial. I hit this myself before too and having to write a dedicated async generator function each time is a bit cumbersome as I think you noted. :-)